File: ime/android/test/Dump.java
import com.cuime.core.CheonjiinCore;import java.util.*;import java.io.*;
public class Dump{public static void main(String[] a)throws Exception{
 PrintWriter w=new PrintWriter(new OutputStreamWriter(new FileOutputStream("/tmp/xl/java.txt"),"UTF-8"));
 for(char c:CheonjiinCore.CHO.toCharArray())for(char v:CheonjiinCore.JUNG.toCharArray())
 for(char j:CheonjiinCore.JONG.toCharArray()){
  String ch=CheonjiinCore.compose(""+c,""+v,(""+j).trim());
  w.println(ch+"\t"+String.join(" ",CheonjiinCore.encode(ch,false)));}
 w.close();}}

File: ime/native/dump.cpp
#include "cheonjiin.hpp"
#include <fstream>
using namespace cuime;
int main(){ std::ofstream o("/tmp/xl/cpp.txt"); const Tables&tb=T();
 for(auto&c:tb.CHO)for(auto&v:tb.JUNG)for(size_t i=0;i<tb.JONG.size();i++){
  std::string j=tb.JONG[i]; if(j==" ")j="";
  std::string ch=compose(c,v,j); o<<ch<<"\t";
  auto e=encode(ch,false); for(size_t k=0;k<e.size();k++){o<<e[k];if(k+1<e.size())o<<" ";}
  o<<"\n";} return 0;}
